blockchain/blocks/src/tipset.rs

```diff
@@ -41,6 +41,14 @@ pub struct Tipset {
     key: OnceCell<TipsetKeys>,
 }
 
+impl TryFrom<&BlockHeader> for Tipset {
+    fn try_from(value: &BlockHeader) -> Result<Self, Error> {
+        Tipset::new(vec![value.clone()])
+    }
+
+    type Error = crate::errors::Error;
+}
+
```

Comment:
Looking at `Tipset::new`, I think the function can only fail if the blocks are incompatible with each other (or if there are zero blocks). So when we're only using one block, creating a tipset cannot fail. Let's change `TryFrom` to `From`.

Reply:
Refactored this